### ISSUE TITLE
Creating an API named `error` results in crash

### ISSUE THREAD

**Issue by mariokandut** [strapi/strapi#12128]
### Describe the bug

Creating an API with the name `error` crashes the hot-reloading of strapi server in development, and starting is only possible if endpoint is deleted.

logs after running `npm run develop`

```
> backend@0.1.0 develop
> strapi develop

[2022-01-07 01:22:05.833] debug: ⛔️ Server wasn't able to start properly.
[2022-01-07 01:22:05.835] error: Cannot read property 'singularName' of undefined
TypeError: Cannot read property 'singularName' of undefined
    at /home/mario/dev/diana2022/backend/node_modules/@strapi/strapi/lib/core/loaders/apis.js:48:21
    at Array.forEach (<anonymous>)
    at validateContentTypesUnicity (/home/mario/dev/diana2022/backend/node_modules/@strapi/strapi/lib/core/loaders/apis.js:47:18)
    at Object.module.exports [as loadAPIs] (/home/mario/dev/diana2022/backend/node_modules/@strapi/strapi/lib/core/loaders/apis.js:36:3)
    at async Strapi.loadAPIs (/home/mario/dev/diana2022/backend/node_modules/@strapi/strapi/lib/Strapi.js:287:5)
    at async Promise.all (index 3)
    at async Strapi.register (/home/mario/dev/diana2022/backend/node_modules/@strapi/strapi/lib/Strapi.js:311:5)
    at async Strapi.load (/home/mario/dev/diana2022/backend/node_modules/@strapi/strapi/lib/Strapi.js:409:5)
    at async Strapi.start (/home/mario/dev/diana2022/backend/node_modules/@strapi/strapi/lib/Strapi.js:161:9)
npm ERR! code 1
npm ERR! path /home/mario/dev/diana2022/backend
npm ERR! command failed
npm ERR! command sh -c strapi develop

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/mario/.npm/_logs/2022-01-07T00_22_05_859Z-debug.log

```

### Steps to reproduce the behavior

Create a new API in the UI with name `error`.

### Expected behavior

There should be a list with blacklisted words when creating an API and a nice hint.

### System

- Node.js version: v15.17.0
- NPM version: 7.4.3
- Strapi version: v.4.0.0

**Comment by kasonde** [comment]
Currently unable to reproduce this with the details provided. 
![image](https://user-images.githubusercontent.com/13610895/148560752-06392138-2d10-424e-bd78-beccaf658693.png)

**Comment by mariokandut** [comment]
@kasonde, I used the strapi starter with next-blog. please try there.

**Comment by kasonde** [comment]
> @kasonde, I used the strapi starter with next-blog. please try there.

Please update that in your issue to make sure anyone else looking at this issue knows exactly what you did to get to his point. You need to share every last detail about your experience as that is what will help us readers help you

**Comment by derrickmehaffy** [comment]
Unsupported node version, also can't reproduce (even with the starter mentioned in comments)

Need more information to reproduce. Marking as closed until more complete information is provided.

Reply back once you have steps needed to reproduce and we can reopen.

**Comment by arielbk** [comment]
I came here from Google since I was also coming across this error, `Cannot read properties of undefined (reading 'singularName')`.

I was trying out some new collection types in my project, but then discarded the changes via git.

This error kept on coming up in the console when I tried to start the project, and it was because git had left behind the empty directories in the `src/api` folder. Removing those directories fixes it.
